src/transaction_builder.cpp

```diff
@@ -156,9 +225,12 @@ TransactionBuilderResult TransactionBuilder::Build()
 
     if (change > 0) {
         // Send change to the specified change address. If no change address
-        // was set, send change to the first Sapling address given as input.
-        if (zChangeAddr) {
-            AddSaplingOutput(zChangeAddr->first, zChangeAddr->second, change);
+        // was set, send change to the first Sapling address given as input,
+        // or the first Sprout address given as input.
```

Comment:
... the first Sapling address given as input if any; otherwise the first Sprout address given as input. (A t-address can only be used as the change address if explicitly set.)